File: Codes/Hash/Non-Cryptographic/ELFHash/ELFHash.vb
'  Modifikasi PJWHash untuk sistem UNIX.
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function ELFHash (key As String) As UInteger
    Dim state As UInteger = 0
	Dim x As UInteger = 0
	Dim i As UInteger = 0

	For i = 0 To key.Length - 1
		state = (state << 4) + CByte(AscW(key(CInt(i))))
		x = state And &HF0000000UI

		If x <> 0 Then
			state = state Xor (x >> 24)
		End If
		state = state And Not x
	Next

	Return state
End Function

File: Codes/Hash/Non-Cryptographic/ELFHash/ELFHash.cs
/*
	Modifikasi PJWHash untuk sistem UNIX.
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint ELFHash (string key)
{
    uint state = 0;
	uint x = 0;
	uint i = 0;

	for (i = 0; i < key.Length; i++)
	{
		state = (state << 4) + ((byte)key[(int)i]);
		
		if ((x = state & 0xF0000000) != 0)
		{
			state ^= (x >> 24);
		}
		
		state &= ~x;
	}

	return state;
}